apps/swirl-docs/src/components/Layout/DocLinksNav.tsx

```diff
@@ -34,20 +34,21 @@ export const DocLinksNav: FunctionComponent<DocLinksNavProps> = ({
           {documentLinkList?.map((link: DocHeadline, index: number) => {
             return (
               <li key={link.id} className="relative font-sm mb-2">
-                <Link
-                  className={`
-                  transition-colors duration-500 ease-in-out
-                  before:transition-colors before:duration-500 before:ease-in-out
-                  before:block before:absolute before:top-0 before:left-[-17px] before:w-[2px] before:h-6 text-sm
-                  ${
-                    currentActiveIndex === index
-                      ? "text-border-info before:bg-border-info"
-                      : "text-text-subdued"
-                  }
-                `}
-                  href={`#${link.id}`}
-                >
-                  <a>{link.name}</a>
+                <Link href={`#${link.id}`}>
+                  <a
+                    className={`
+                    transition-colors duration-500 ease-in-out
+                    before:transition-colors before:duration-500 before:ease-in-out
+                    before:block before:absolute before:top-0 before:left-[-17px] before:w-[2px] before:h-6 text-sm
+                    ${
+                      currentActiveIndex === index
+                        ? "text-border-info before:bg-border-info"
+                        : "text-text-subdued"
+                    }
+                  `}
+                  >
+                    {link.name}
```

Comment:
**issue:** Something seems off with the labels here.
<img width="1051" alt="Bildschirm­foto 2022-11-09 um 10 17 23" src="https://user-images.githubusercontent.com/1019414/200790461-d0ce9373-1419-494c-8b5b-03f76e56b372.png">
<img width="1078" alt="Bildschirm­foto 2022-11-09 um 10 17 49" src="https://user-images.githubusercontent.com/1019414/200790466-a6fb92a2-793e-452b-be9b-4061ff6ad3c2.png">

Reply:
Yeah it seems the observer does not work correctly anymore with that many headings. I would create a seperate bug ticket for that.

Reply:
will be resolved in this [ticket](https://flipapp.myjetbrains.com/youtrack/issue/APP-12197/SWIRL-DOCS-Scroll-Observer-has-bugs-on-big-sites-and-headings-with-same-name)